uaclient/entitlements/livepatch.py

```diff
@@ -78,7 +78,14 @@ def enable(self):
                 util.subp(['apt-get', 'install', '--assume-yes', 'snapd'],
                           capture=True)
             print('Installing canonical-livepatch snap...')
-            util.subp(['snap', 'install', 'canonical-livepatch'], capture=True)
+            try:
+                util.subp(['snap', 'install', 'canonical-livepatch'],
+                          capture=True)
+            except util.ProcessExecutionError as e:
+                msg = 'Unable to install Livepatch client: ' + str(e)
+                print(msg)
+                logging.error(msg)
+                return False
```

Comment:
snap install is network dependent and sometimes was flaky (at least for me) in the past. Would we want to retry this a few times before giving up dumping the exception to the user?

Reply:
I wondered this same thing, but we have numerous network dependent items (e.g. apt update, apt install, etc.) that are not wrapped either. I'd say the user for now should retry the command.

Reply:
Ok, I'll open an issue then to reconsider improving that in the long run